src/Nethermind/Nethermind.TxPool/TransactionExtensions.cs

```diff
@@ -4,14 +4,25 @@
 using System.Runtime.CompilerServices;
 using Nethermind.Core;
 using Nethermind.Int256;
+using Nethermind.Serialization.Rlp;
 
 [assembly: InternalsVisibleTo("Nethermind.TxPool.Test")]
 
 namespace Nethermind.TxPool
 {
-    internal static class TransactionExtensions
+    public static class TransactionExtensions
     {
-        public static UInt256 CalculateGasPrice(this Transaction tx, bool eip1559Enabled, in UInt256 baseFee)
+        private static readonly int MaxSizeOfTxForBroadcast = 128_000; //128KB, as proposed in https://eips.ethereum.org/EIPS/eip-5793
+        private static readonly ITransactionSizeCalculator _transactionSizeCalculator = new TxDecoder();
+
+        public static int GetLength(this Transaction tx)
+        {
+            return tx.GetLength(_transactionSizeCalculator);
```

Comment:
Maybe we can remove `GetLength` from transaction class and leave it as extension? And probably remove `ITransactionSizeCalculator` interface?

Reply:
Why it wasn't working? This sounds like a good idea.